### PR TITLE
update type declarations of Schema.index method

### PR DESCRIPTION
**Pull request by Raader**
added proper types to arguments of index method on Schema class.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1231,6 +1231,8 @@ declare module 'mongoose' {
   type PreSaveMiddlewareFunction<T> = (this: T, next: (err?: CallbackError) => void, opts: SaveOptions) => void | Promise<void>;
   type PostMiddlewareFunction<ThisType, ResType = any> = (this: ThisType, res: ResType, next: (err?: CallbackError) => void) => void | Promise<void>;
   type ErrorHandlingMiddlewareFunction<ThisType, ResType = any> = (this: ThisType, err: NativeError, res: ResType, next: (err?: CallbackError) => void) => void;
+  
+  type Index = { [key: string]: boolean | number | string | Index }
```
Comment (vkarpov15) [inline index.d.ts:1235]: Why is this `| Index` here? I don't know of any recursive index definitions.

Comment (Raader) [reply index.d.ts]: My bad, changed it to mongodb's index type, It's my first ever pr on a project like this. it is not much but it should help with intellisense.

**Review by vkarpov15 — APPROVED** [review]
Thanks :+1:

**Comment by Gotos** [comment]
This seems to break index creation. `mongodb.IndexSpecification` doesn't seem to be the right type for the fields parameter. According to the docu and how we used mongoose so far, the fields are "just" an object with the fields that you want to use for the index as keys and values for those set to 1. IndexSpecification is vastly different from that. I think this is likely an error? Or am I missing something?

**Comment by Gotos** [comment]
Nevermind, I just realized this was fixed in 5.13.7. Sorry!

**Pull request merged**